lib/email/utils/helpers.js

```diff
@@ -63,7 +63,18 @@ function logErrorIfHeadersAreWeirdOrMissing (log, message, origin) {
   const headers = getHeaders(message)
   if (headers) {
     const type = typeof headers
-    if (type !== 'object') {
+    if (type === 'object') {
+      const deviceId = getHeaderValue('X-Device-Id', message)
+      const uid = getHeaderValue('X-Uid', message)
+      if (! deviceId && ! uid) {
```

Comment:
For my immediate purposes, it's sufficient to only warn if both `X-Device-Id` and `X-Uid` are unset at the moment. But we could expand the test to include other properties too if it's useful, I'm thinking about e.g. `X-Flow-Id`.